airflow/jobs/scheduler_job_runner.py

```diff
@@ -2044,9 +2044,12 @@ def _cleanup_stale_dags(self, session: Session = NEW_SESSION) -> None:
         session.flush()
 
     def _set_orphaned(self, dataset: DatasetModel) -> int:
-        self.log.info("Orphaning unreferenced dataset '%s'", dataset.uri)
-        dataset.is_orphaned = expression.true()
-        return 1
+        if not dataset.is_orphaned:
+            self.log.info("Orphaning unreferenced dataset '%s'", dataset.uri)
+            dataset.is_orphaned = expression.true()
+            return 1
+
+        return 0
```

Comment:
Nit, think positively:
```suggestion
        if dataset.is_orphaned:
            return 0
        self.log.info("Orphaning unreferenced dataset '%s'", dataset.uri)
        dataset.is_orphaned = expression.true()
        return 1
```